tests/gis_tests/gdal_tests/test_geom.py

```diff
@@ -689,7 +689,7 @@ def test_geometry_types(self):
             ("Triangle M", 2017, False),
             ("Point ZM", 3001, True),
             ("LineString ZM", 3002, True),
-            ("Polygon ZM", 3003, False),
+            ("Polygon ZM", 3003, True),
```

Comment:
@smithdc1 Are you going to send another 11 PRs for all `ZM` geometry types? :upside_down_face: just asking :smiling_face_with_tear:

Reply:
Good question!

The remaining items that are currently supported are all variations of `GeometryCollection` so was going to do that in one go. I'm not expecting there to be any API changes with those, they should 'just work'™️ 

As I'm expecting those to be similar to Polygon's (that is, just adding them to `GEO_CLASSES`) shall I add that as part of this PR? 

Then there's new types (circular geometries) which I think is a separate PR as that'll add new bindings to the c library. I'm preping a larger PR for that then we can discuss if we merge that in chunks like with #17696

Reply:
> As I'm expecting those to be similar to Polygon's (that is, just adding them to `GEO_CLASSES`) shall I add that as part of this PR?

Already merged :sweat_smile: Please submit the next one.